app/services/idv/analytics_events_enhancer.rb

```diff
@@ -50,12 +50,37 @@ def self.included(_mod)
     def common_analytics_attributes
       {
         proofing_components: proofing_components,
-      }
+      }.merge({
+        sp_request: sp_request_components,
+      }.compact)
     end
 
     def proofing_components
       return if !user&.respond_to?(:proofing_component) || !user.proofing_component
       ProofingComponentsLogging.new(user.proofing_component)
     end
+
+    def sp_request_components
```

Comment:
I think I got my wires crossed when I said our tickets were related--in refinement we decided that this should be in the Analytics service, not the AnalyticsEventsEnhancer

Reply:
Yeah, I definitely mis-read the ticket. Didn't realize it was for *all* events. Will update.

Reply:
this is going to add a lot of keys to every single log event, have we done any rough estimates for how many KB this would add to logs? (I only ask because at a certain point, CW stops parsing JSON payloads so I want to get a sense of how many events this would push past "the edge")

Reply:
Looks like max event size is 256kb (presumably that's serialized JSON?) and Cloudwatch "can discover a maximum of 1000 log event fields in a log group," according to [this page](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html). Presumably the 1000 log event fields limit is what makes it seem like they just give up parsing after a while.

I can think of a couple of optimizations here:

1. Exclude the `sp_request` key entirely if there is no associated SP
2. Only include keys on `sp_request` where the value is `true`
3. Use `1` instead of `true` to represent values
4. Log component values as e.g. `{ C1: 1, C2: 1, P1: 1 }` instead of `[{name: "C1"}, {name: "C2"}, {name: "P1"}]`

Reply:
ooo the `1` for true seems neat, but since it doesn't affect the number of fields, prob easier to leave as a boolean. I like the key-value approach in (4) because that would make it very easy to query for "all requests with C1" vs having to search the array

Reply:
In 51fb04ac5, I only add keys that are true. For legacy IAL1, that results in just the component_values. Does that sound reasonable?

Reply:
I I posted my previous comment before I saw Matt's comment. In 070b5f935 I also convert the component values to a flat hash. This works great for requests with a VoT but has the unfortunate side effect for legacy requests (i.e. all current requests) of looking like this:
```
component_values: { 'http://idmanagement.gov/ns/assurance/ial/1' => 1 }
```

Reply:
In 012036826 I pick up Zach's cleaner conversion of the component_values. At this point we're doing 1, 2, and 4 of Matt's suggestions above.

Reply:
Would it be wrong to chomp "http://idmanagement.gov/ns/assurance/" off the start of acr values (i.e. cut it down to 'ial/1')? I think it would still allow us to get the data we need while keeping log entries a little more svelte

Reply:
Alternately replace it with a US flag emoji, i.e. "🇺🇸/ial/1" 😆